supervisor/addons/addon.py

```diff
@@ -395,7 +395,7 @@ def ingress_port(self) -> int | None:
 
         port = self.data[ATTR_INGRESS_PORT]
         if port == 0:
-            return self.sys_ingress.get_dynamic_port(self.slug)
+            _LOGGER.critical("No port set for add-on %s", self.slug)
```

Comment:
This isn't right. If this code is actually called in practice things are going to go poorly. The caller will get back `0` and not realize anything is wrong since no exception was raised and the user won't have a clue what to do with this log. I think we have to take one of the following options:

1. Change from a `property` to `async def get_ingress_port` and adjust callers accordingly. Code inside remains the same other then awaiting `get_dynamic_port`
2. This property can return `int | None | asyncio.Task`. Change this code to `return self.sys_create_task(self.sys_ingress.get_dynamic_port(self.slug))`. Callers will need to be adjusted to await the task if that is what they are returned
3. This code raises an exception here. Callers will need to be adjusted to catch that exception and handle it by doing `await self.sys_ingress.get_dynamic_port(addon.slug)`

1 feels the most clean to me even if its not a property. I like 3 the least because it puts this work and knowledge of how to get an ingress port on the caller when it feels like it should be part of the `Addon` class.

Reply:
1. That was what I had previously, but based on discussions above I changed it to get initialized earlier to avoid making this a get method (see https://github.com/home-assistant/supervisor/pull/4677#discussion_r1391436425).
2. Seems quite an awkward implementation and puts burden on the caller.
3. Similarly.

I agree that if we want dynamic port allocation when this property is used, then 1 is the cleanest solution.

But we took the direction to allocate statically at installation time. With the new implementation, this if statement is more sanitary. My intention was to get it to show up in Sentry in case, contrary to all expectations, it still happens. Maybe better to just throw an exception?

Reply:
Ok I see pascal's point. In that case I feel like we should probably just make it a fatal exception if its unset rather then returning 0. If it should always be set its ok (and probably preferable) if it fails catastrophically if somehow it happens to get in that state. Similar to how all the properties work in `CoreSys`:
https://github.com/home-assistant/supervisor/blob/176b63df52c54ccccf47c21144a22f271b5e61b4/supervisor/coresys.py#L320-L322

Reply:
Ok, I've made it a runtime error.